src/core/router/tunnel/impl.cc

```diff
@@ -668,13 +668,13 @@ void Tunnels::ManageInboundTunnels() {
 
 void Tunnels::ManageTransitTunnels() {
   std::uint64_t ts = kovri::core::GetSecondsSinceEpoch();
+
+  std::unique_lock<std::mutex> l(m_TransitTunnelsMutex);
   for (auto it = m_TransitTunnels.begin(); it != m_TransitTunnels.end();) {
     if (ts > it->second->GetCreationTime() + TUNNEL_EXPIRATION_TIMEOUT) {
       auto tmp = it->second;
-      LOG(debug) << "Tunnels: transit tunnel " << tmp->GetTunnelID() << " expired"; {
-        std::unique_lock<std::mutex> l(m_TransitTunnelsMutex);
-        it = m_TransitTunnels.erase(it);
-      }
+      LOG(debug) << "Tunnels: transit tunnel " << tmp->GetTunnelID() << " expired";
+      it = m_TransitTunnels.erase(it);
       delete tmp;
```

Comment:
Unrelated to your changes, but since we're here, could you also remove the manual delete here? IIUC, deleting after invalidating the reference with `m_TransitTunnels.erase(it)` leads to a delete-after-free bug.

If I'm correct, you should also remove `auto tmp = it->second;`, since it would be unnecessary.

Reply:
>IIUC, deleting after invalidating the reference with m_TransitTunnels.erase(it) leads to a delete-after-free bug

Yes, invalid delete. Valgrind will confirm.

>If I'm correct, you should also remove auto tmp = it->second;, since it would be unnecessary.

Yes, but replace the copy with `it->second` so we can keep the log.

Reply:
Referencing #511.